File: src/main/java/xyz/oribuin/skyblock/command/impl/WarpCommand.java
package xyz.oribuin.skyblock.command.impl;

import dev.rosewood.rosegarden.RosePlugin;
import dev.rosewood.rosegarden.command.framework.ArgumentsDefinition;
import dev.rosewood.rosegarden.command.framework.BaseRoseCommand;
import dev.rosewood.rosegarden.command.framework.CommandContext;
import dev.rosewood.rosegarden.command.framework.CommandInfo;
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable;

public class WarpCommand extends BaseRoseCommand {

    public WarpCommand(RosePlugin rosePlugin) {
        super(rosePlugin);
    }

    @RoseExecutable
    public void execute(CommandContext context) {
        // TODO: Open the Warps GUI
    }

    @Override
    protected CommandInfo createCommandInfo() {
        return CommandInfo.builder("warp")
                .descriptionKey("command-warp-description")
                .permission("skyblock.warp")
                .playerOnly(true)
                .build();
    }

    @Override
    protected ArgumentsDefinition createArgumentsDefinition() {
        return ArgumentsDefinition.builder().build();
    }

    //     xyz.oribuin.skyblock.command.impl.sub.warp.WarpCategoryCommand::class.java,
    //    xyz.oribuin.skyblock.command.impl.sub.warp.WarpIconCommand::class.java,
    //    xyz.oribuin.skyblock.command.impl.sub.warp.WarpNameCommand::class.java,
    //    xyz.oribuin.skyblock.command.impl.sub.warp.WarpSettingsCommand::class.java,
    //    xyz.oribuin.skyblock.command.impl.sub.warp.WarpTeleportCommand::class.java

}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/command/command/WarpCommand.kt
package xyz.oribuin.skyblock.command.command

import dev.rosewood.rosegarden.RosePlugin
import dev.rosewood.rosegarden.command.framework.CommandContext
import dev.rosewood.rosegarden.command.framework.RoseCommand
import dev.rosewood.rosegarden.command.framework.RoseCommandWrapper
import dev.rosewood.rosegarden.command.framework.RoseSubCommand
import dev.rosewood.rosegarden.command.framework.annotation.Optional
import dev.rosewood.rosegarden.command.framework.annotation.RoseExecutable
import xyz.oribuin.skyblock.command.command.sub.warp.*
import xyz.oribuin.skyblock.gui.WarpsGUI
import xyz.oribuin.skyblock.util.asMember
import xyz.oribuin.skyblock.util.getMenu

class WarpCommand(rosePlugin: RosePlugin, parent: RoseCommandWrapper) : RoseCommand(
    rosePlugin, parent,
    WarpCategoryCommand::class.java,
    WarpIconCommand::class.java,
    WarpNameCommand::class.java,
    WarpSettingsCommand::class.java,
    WarpTeleportCommand::class.java
) {

    @RoseExecutable
    fun execute(context: CommandContext, @Optional type: RoseSubCommand?) {
        this.rosePlugin.getMenu(WarpsGUI::class).openMenu(context.asMember(this.rosePlugin))
    }

    override fun getDefaultName(): String = "warp"

    override fun getDescriptionKey(): String = "command-warp-description"

    override fun getRequiredPermission(): String = "skyblock.command.warp"

    override fun isPlayerOnly(): Boolean = true

}